### ISSUE TITLE
Consider a 'slideshow' directive based on JSAV

### ISSUE THREAD

**Issue by DaveParillo** [RunestoneInteractive/RunestoneComponents#532]
The github project vkaravir/JSAV provides a capability that could augment both codelens and matrixeq.

Originally developed for OpenDSA, which is another open source sphinx-based software textbook authoring system.

JSAV allows the creation of static images, arrays, linked lists, trees, matrices, and graph structures all using a simple javascript api.

Codelens is sometimes not the most elegant visualization even for python - it can be large an spread out awkwardly.  For non-python, the support is experimental - most simple examples involving  pointers in C and C++ do not work at all.

JSAV provides a means to provide step-by-step walkthroughs more under the control of the author and also allow a means to visualize matrix operations.

**Comment by bnmnetp** [comment]
Yeah, I've talked with the OpenDSA guys before, and even co-authored a paper on eBooks with some of them.

If we can find a reasonable strategy to add directives for JSAV into the runestone package I'm all for it.

**Comment by DaveParillo** [comment]
I used to use OpenDSA, but stopped because the builds are too far removed from the docutils/ sphinx style. It's trying to do too many things at once and authoring in that environment was just too painful for me.

JSAV scripts are essentially a block of javascript.  In a certain sense, it's not too different from the existing sphinx.ext.graphviz package, which I use a lot.

How to bind the code in a directive to the underlying JSAV without things getting all ugly is a good question.
